### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by framillien**
### Remove additionnal SSH lib

JCraft Jsch lib is the main SSH lib used.
J2SSH is only used for a passphrase validity check.
Replace J2SSH by Jsch calls to implement validity check.

### Allow packaging with JDK 9, updating tests lib

Previous PowerMock implementation is not compatible with Java9, upgrade PowerMock, Mockito & co to gain this compatibility.

Do needed changes (null parameter is no more matched by any() matchers) and try to use same patterns for static Mock imports:
- By default import static methods of Mockito
- Import harmcrest static matchers when necessary
- Never import static method of PowerMock: We need to know when we use PowerMock to limit his usage

### CODE REVIEW

File: src/main/java/com/sonymobile/tools/gerrit/gerritevents/ssh/SshUtil.java [hunk 1]
```diff
@@ -42,53 +42,26 @@
     private SshUtil() {
     }
 
-    /**
-     * Does a series of checks in the file to see if it is a valid private key file.
-     * @param keyFile the file
-     * @return true if valid.
-     */
-    public static boolean isPrivateKeyFileValid(File keyFile) {
```
Comment (rsandell) [inline src/main/java/com/sonymobile/tools/gerrit/gerritevents/ssh/SshUtil.java]: removal of public method signature.

File: README.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ All event types can be found in the [com.sonymobile.tools.gerrit.gerritevents.dt
     * `java-1.6`
         * `maven-3.0.4`
 
+* Java 8 & 9: Works.
```
Comment (rsandell) [inline README.md]: Btw how do you know if you didn't change the compiler setting in the pom?

Comment (framillien) [reply README.md]: Hi, this change allow us to run tests on Java 9 runtime. Code is still targeting Java 6 (enforced in POM) but 'mvn tests' can be executed on Java 9.

Without this changes tests are working only with a Java 8 runtime. Tested on macos with:
/Library/Java/JavaVirtualMachines/jdk1.8.0_162.jdk
/Library/Java/JavaVirtualMachines/jdk-9.0.1.jdk/

Comment (rsandell) [reply README.md]: I updated the pom to compile with Java 8 after the merge :)

**Pull request merged**